src/parse.ts

```diff
@@ -28,9 +28,9 @@ export interface ParsedHost {
  * @returns A parsed URL object.
  */
 export function parseURL(input = "", defaultProto?: string): ParsedURL {
-  const dataMatch = input.match(/^(data:|blob:)/);
+  const dataMatch = input.match(/^(blob|data|javascript|vbscript):/);
```

Comment:
Maybe we can trimLeft before instead of updating regexes?

Reply:
what do you mean?

Reply:
input.trimStart() instead of updating protocol regex

Reply:
I don't think that would resolve the issue. The later regexp _requires_ `//` in a protocol to match it.

Reply:
Will check it locally but from what i see we are adding leading space to protocol regex (which makes an invalid input marked as having valid protocol).

Reply:
whitespace seems to be allowed within protocols

Reply:
but we're not allowing it _here_ - all this line does is add more protocols

(This PR contains two separate fixes.)

Reply:
Updated with an additional test. It is interesting `URL` constructor allows passing a URL starting with space/zero char. But it also **removes** the leading spaces as well basically `new URL("\0https://google.com").protocol === "https:"` is true (while including in regex, means we match protocol as `\0https:`)

I am looking more into spec. I think it is fine if we do same normalization for URL parser as finally parsed output is safe and sound.

But i am worried about giving the behavior to `hasProtocol("\0https://)` returning `true`. It can open other security issues (it is safe only when user after `hasProtocol` check passing the input `URL` constructor (even `ufo.normalizeURL` seems not to handle such case currently)

Reply:
The issue is more that if we say there is _not_ a protocol it can be much more unsafe as there can be genuine security issues

Reply:
I would love to see such issues reported before fixing for `hasProtocol` :)

Reply:
(pushed fix for other spec)

Reply:
Seems URL (for merging also considers that)

<img width="715" alt="image" src="https://github.com/unjs/ufo/assets/5158436/4da4df85-c578-426f-9545-b54d9efda397">